src/js/components/modals/ServiceFormModal.js

```diff
@@ -135,6 +136,13 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let {state, props} = this;
+    return props.open !== nextProps.open ||
+      state.jsonMode !== nextState.jsonMode ||
+      state.errorMessages !== nextState.errorMessage;
+  }
```

Comment:
the errorMessage gets set to null on change (reads: focus/blur/change) thus we have to re-render.